### DISCUSSION TITLE
Built-in HELICS apps

### DISCUSSION THREAD

**Discussion by allisonmcampbell** [GMLC-TDC/HELICS#1578] [Q&A] [answered]
I'm in the process of revising the structure of the user guide docs, and coming up against some confusion over the correct home for the built-in HELICS apps:
https://docs.helics.org/en/v3userguide/apps/index.html

To help resolve this, I have a few questions:
1. What do you personally use these apps for?  
2. Which apps do you use most frequently?
3. When is the last time you used the apps?

**Comment by phlptp** [comment, marked as answer]
aside from the standalone broker app.

I have used the recorder/clone and player.
Basically use the recorder to capture the output from a federate, and then the player to either inject some signals or actually play back the recording.  Occasionally use the clone app to duplicate all of a federate, to create a mock.  

the apps gets used frequently as part of the testing but probably a couple weeks since I last used them outside of the testing system.

**Reply by phlptp** [reply]
Use recorder to capture specific components,  clone for copying a federate for making a mock, and player for playback, signal injection, or to become the mock.

**Reply by trevorhardy** [reply]
I like @phlptp's answer as it's similar to mine; I'm going to mark it as "the answer" since it looks like things have stalled out here.

**Comment by nightlark** [comment]
I mostly use the broker and sometimes helics_app (for running the echo/player apps as test federates).
